pinot-plugins/pinot-file-system/pinot-s3/pom.xml

```diff
@@ -183,27 +183,9 @@
     </dependency>
     <dependency>
       <groupId>com.adobe.testing</groupId>
-      <artifactId>s3mock-testng</artifactId>
+      <artifactId>s3mock-testcontainers</artifactId>
```

Comment:
`s3mock-testcontainers` is [recommended](https://github.com/adobe/S3Mock#s3mock-java) as it avoids bring in deps from s3mock transitively, which brings in the spring framework and also requires some [specific keystore](https://github.com/adobe/S3Mock/issues/720). 

The new version of s3mock is more accurate on handling the leading and trailing slash in s3 object names, so the changes on the unit tests as made in this PR.